### DISCUSSION TITLE
Specifying number of speakers in the Huggingface Inference API

### DISCUSSION THREAD

**Discussion by Jeannotisintheplace** [pyannote/pyannote-audio#1088] [Q&A]
Hi team,

Could you please explain how to pass the known-in-advance number of speakers to the API ?

Shall "num_speaker" be specified in the Headers (or parameters?), and in which format ?

Sorry for possible trivial question but thanks in advance,

Julien

**Comment by hbredin** [comment]
I do not think this is supported. 

I guess it should be possible as there are tasks for which options can be passed to the API:
https://huggingface.co/docs/api-inference/detailed_parameters#fill-mask-task

This part is handled outside of pyannote in Huggingface repo: https://github.com/huggingface/api-inference-community/tree/main/docker_images/pyannote_audio

cc @julien-c who might be able to indicate what change should be made to add this feature to the API.

This is more or less what is run on Huggingface servers:

```python
pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization")
pipeline("audio.wav")
```

This is what @Jeannotisintheplace would like to run:

```python
pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization")
pipeline("audio.wav", num_speakers=2)
```
